File: shared-atlas-dart/lib/src/interfaces/yjs_sync_payload.dart
class YjsSyncPayload {
  final String noteId;
  final String update;
  final String stateVector;

  YjsSyncPayload({
    required this.noteId,
    required this.update,
    required this.stateVector,
  });

  factory YjsSyncPayload.fromJson(Map<String, dynamic> json) {
    return YjsSyncPayload(
      noteId: json['noteId'] as String,
      update: json['update'] as String,
      stateVector: json['stateVector'] as String,
    );
  }
  Map<String, dynamic> toJson() {
    return {
      'noteId': this.noteId,
      'update': this.update,
      'stateVector': this.stateVector,
    };
  }
}

File: packages/shared-atlas-rust/src/interfaces/yjs_sync_payload.rs
use serde::{Serialize, Deserialize};

/// Payload for Y.js initial sync.
/// Y.js 初始同步的 Payload。
/// 
/// Event: `server.emit('yjs:sync', payload)`
#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct YjsSyncPayload {
/// Note ID / 笔记ID
    pub note_id: String,
/// Y.js update data (base64) / Y.js 更新数据
    pub update: String,
/// State vector (base64) / 状态向量
    pub state_vector: String,
}